addon/adapters/pouch.js

```diff
@@ -0,0 +1,161 @@
+import Ember from 'ember';
+import DS from 'ember-data';
+import PouchDB from 'pouchdb';
+
+import {
+  extractFindOne,
+  extractDeleteRecord,
+  updateSchemaForType
+} from 'ember-pouch/utils';
+
+var bind = Ember.run.bind;
+var next = Ember.run.next;
+
+export default DS.Adapter.extend({
+  defaultSerializer: '-pouch',
+  coalesceFindRequests: true,
+  schema: null,
+  url: null,
+
+  setup: function(url) {
+    url = url || this.url;
+
+    if (url) { this.db = new PouchDB(url); }
+  }.on('init'),
```

Comment:
Unfortunately that defeats the point of `super` as it is already an optional call. Subclassing is a refinement on the ancestor classes, when refining a methods it is the choice of the author to fully replace, or merely refine parts of the inherited methods functionality. Super prevents the need to always fully re-implement when refining.